crates/polars-plan/src/utils.rs

```diff
@@ -370,6 +370,24 @@ pub(crate) fn aexprs_to_schema(
         .collect()
 }
 
+/// Concatenate multiple schemas into one, disallowing duplicate field names
+pub fn merge_schemas(schemas: &[SchemaRef]) -> PolarsResult<Schema> {
+    let schema_size = schemas.iter().map(|schema| schema.len()).sum();
+    let mut merged_schema = Schema::with_capacity(schema_size);
+
+    for schema in schemas {
+        schema.iter().try_for_each(|(name, dtype)| {
+            if merged_schema.with_column(name.clone(), dtype.clone()).is_none() {
```

Comment:
Not a blocker for this PR, but this can be done elegantly with `ok_or_else(|| polars_err!(..)`.

Reply:
FWIW, this is actually doing the opposite of `ok_or_else`, a `Some` result should be mapped to an `Err` and a `None` to `Ok(())`. I couldn't find a more concise way to express this.

Reply:
Oh right!